### DISCUSSION TITLE
Pico usb access to config file

### DISCUSSION THREAD

**Discussion by SingleBladePaddler** [micropython/micropython#10374] [Using MicroPython]
First post on here, quite new to this.
I wondered if there is a way of allowing a pico to be accessed as a usb drive with micropython installed. to allow setting for SSID and password etc. I did ponder having serial port opened on the device and using puTTY or such to allow setting changes.  Any thought or recommendations.

**Comment by fivdi** [comment]
Although it's not the same as a USB drive, I think you should be able to achieve what you would like to achieve with the mpremote utility. mpremote will allow you to copy files to the Pico.

See
https://docs.micropython.org/en/latest/reference/mpremote.html
https://pypi.org/project/mpremote/

**Reply by SingleBladePaddler** [reply]
Thanks for the prompt reply. I'll have look and play with that

**Comment by cve2022** [comment]
Have a look at [rshell](https://github.com/dhylands/rshell) ([https://pypi.org/project/rshell/](https://pypi.org/project/rshell/)). The pico fs appears as /pyboard and you may easily copy files from/to it (among other things).

**Reply by SingleBladePaddler** [reply]
Thanks I have looked at rshell and was very straight forward to install

**Reply by cve2022** [reply]
Good to know it worked. :) I assume you are developing on Linux. On Windows 10 there is a caveat, the default installation with pip does not work, it requires installing a different dependency from the standard (pyreadline3 instead of pyreadline installed by default by pip).

**Comment by mattytrentini** [comment]
USB mass storage is possible, though it's disabled by default; you'll have to build it yourself. There are a number of caveats with USB, see #7402 for some of the details.

**Comment by SingleBladePaddler** [comment]
Thanks for the replies. Looking at the micropython manual under files systems there is function within micropython for the STM32 to create a hybrid system with fat partition and a littleFS partition. looks ideal. not sure if i have the skill level to convert the pyb for the rp2040.
